WolvenKit.App/ViewModels/GraphEditor/RedGraph.Scene.cs

```diff
@@ -338,26 +338,35 @@ public static RedGraph GenerateSceneGraph(string title, scnSceneResource sceneRe
         var graph = new RedGraph(title, sceneResource);
 
         var nodeCache = new Dictionary<uint, BaseSceneViewModel>();
-        foreach (var nodeHandle in sceneResource.SceneGraph.Chunk!.Graph)
+        var sceneGraph = sceneResource.SceneGraph?.Chunk!.Graph;
+        if (sceneGraph != null)
         {
-            ArgumentNullException.ThrowIfNull(nodeHandle.Chunk);
 
-            var node = nodeHandle.Chunk;
+            foreach (var nodeHandle in sceneGraph)
+            {
+                ArgumentNullException.ThrowIfNull(nodeHandle.Chunk);
 
-            var nvm = graph.WrapSceneNode(node);
+                var node = nodeHandle.Chunk;
 
-            if (!nodeCache.ContainsKey(nvm.UniqueId))
-            {
-                nodeCache.Add(nvm.UniqueId, nvm);
-                graph.Nodes.Add(nvm);
+                var nvm = graph.WrapSceneNode(node);
 
-                graph._currentSceneNodeId = Math.Max(graph._currentSceneNodeId, nvm.UniqueId);
-            }
-            else
-            {
-                _loggerService?.Warning("Duplicate node ID: " + nvm.UniqueId.ToString() + ". Some nodes may be missing in graph view. File: " + title);
+                if (!nodeCache.ContainsKey(nvm.UniqueId))
+                {
+                    nodeCache.Add(nvm.UniqueId, nvm);
+                    graph.Nodes.Add(nvm);
+
+                    graph._currentSceneNodeId = Math.Max(graph._currentSceneNodeId, nvm.UniqueId);
+                }
+                else
+                {
+                    _loggerService?.Warning("Duplicate node ID: " + nvm.UniqueId.ToString() + ". Some nodes may be missing in graph view. File: " + title);
+                }
             }
         }
+        else
+        {
+            _loggerService?.Error("Scene graph failed to initialize. Check that the sceneGraph is created.");
+        }
 
         foreach (var node in graph.Nodes)
```

Comment:
Does it even make sense to initialize the graph if the sceneGraph is invalid? (Genuine question, because I don't know.)
If not, you could maybe `return graph` in the `if sceneGraph is null` case and save a bunch of `else`s